packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -322,9 +322,10 @@ export const adjustSectionValue = <TDate extends PickerValidDate, TSection exten
     }
 
     const currentOptionIndex = options.indexOf(section.value);
-    const newOptionIndex = (currentOptionIndex + options.length + delta) % options.length;
+    const newOptionIndex = (currentOptionIndex + delta) % options.length;
+    const adjustedIndex = (newOptionIndex + options.length) % options.length;
```

Comment:
My first thought was to use Math.abs(), which I believe could also work as a fix, but after some thinking I concluded that this approach would be safer to assure that we will always return an index within the correct range.

Reply:
I do agree that using the modulo seems safer than an absolute value if in the future for some reason we end up with a weird delta.

Nitpick on the naming, since the operation we are doing in a clamping (https://en.wikipedia.org/wiki/Clamping_(graphics))

```suggestion
    const clampedIndex = (newOptionIndex + options.length) % options.length;
```